packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -1094,7 +1094,12 @@ Environment variables can be marked for removal when used in Lambda@Edge by sett
       throw new Error('Cannot configure \'securityGroup\' or \'allowAllOutbound\' without configuring a VPC');
     }
 
-    if (!props.vpc) { return undefined; }
+    if (!props.vpc) {
+      if (props.vpcSubnets) {
```

Comment:
```suggestion
    if (!props.vpc && props.vpcSubnets) {
```

Reply:
I thought about doing this, but there's the `return undefined;` after the inner `if`. So it'd have to be rewritten as:

```ts
if (!props.vpc && props.vpcSubnets) {
  throw new Error("...");
} else if (!props.vpc) {
  return undefined
}
```

which felt a little messy and maybe more confusing to repeat half of the condition?

Actually mean to include a note/question about this when I pushed the new commit but got distracted by the failing build 😄 So I'm happy to making the change but just want to make sure that this is the desired outcome?

Reply:
No need for you to revise because of a small stylistic difference. I only made the suggestion because I was also making the suggestion to change the error message. Otherwise, I would have just plain approved since @corymhall is also in favor.

Reply:
But also, I didn't notice that the error message above was infinitely better than the one I suggested. So, sorry, one more suggested change to that.

Reply:
All good! And I meant "meant to" instead of "mean to" -- oops, sorta changes the tone.  But in any case, thanks for the feedback on the error message. I read https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md#error-messages and I think ended up writing something too wordy; this one is much clearer!